addons/mail/static/src/new/discuss/autogrow_input.js

```diff
@@ -0,0 +1,97 @@
+/** @odoo-module */
+
+import { useFocus, useHover } from "../utils";
+import {
+    Component,
+    useRef,
+    useState,
+    useExternalListener,
+    onWillUpdateProps,
+} from "@odoo/owl";
+
+export class AutogrowInput extends Component {
+    setup() {
+        this.state = useState({
+            value: this.props.value,
+        });
+        this.inputFocus = useFocus("input");
+        this.inputRef = useRef("input");
+        this.rootHover = useHover("root");
+        onWillUpdateProps((nextProps) => {
+            if (this.props.value !== nextProps.value) {
+                this.state.value = nextProps.value;
+            }
+        });
+        useExternalListener(
+            document,
+            "click",
+            (ev) => {
+                if (this.inputRef && !this.inputRef.el.contains(ev.target)) {
+                    this.onValidate();
+                }
+            },
+            { capture: true }
+        );
+    }
+
+    /**
+     * @param {KeyboardEvent} ev
+     */
+    onKeydownInput(ev) {
+        switch (ev.key) {
+            case "Enter":
+                this.onValidate();
+                this.inputRef.el.blur();
+                break;
+            case "Escape":
+                this.onDiscard();
+                this.inputRef.el.blur();
+                break;
+        }
+    }
+
+    onValidate() {
+        this.props.onValidate({
+            value: this.inputRef.el.value,
+        });
+        this.state.value = this.props.value;
+    }
+
+    onDiscard() {
+        this.props.onDiscard({
+            value: this.inputRef.el.value,
+        });
+        this.state.value = this.props.value;
```

Comment:
Makes more sense to use `this.state.value` rather than `this.inputRef.el.value`.